operator/internal/controller/redpanda/redpanda_controller_test.go

```diff
@@ -455,6 +456,58 @@ func (s *RedpandaControllerSuite) TestClusterSettings() {
 	s.deleteAndWait(rp)
 }
 
+func (s *RedpandaControllerSuite) TestLicenseReal() {
+	license := os.Getenv("REDPANDA_SAMPLE_LICENSE")
+	if license == "" {
+		s.T().Skip("License environment variable is not provided")
+	}
+
+	rp := s.minimalRP()
+	rp.Spec.ClusterSpec.Statefulset.PodTemplate = &redpandav1alpha2.PodTemplate{
+		Spec: &redpandav1alpha2.PodSpecApplyConfiguration{
+			PodSpecApplyConfiguration: &applycorev1.PodSpecApplyConfiguration{
+				Containers: []applycorev1.ContainerApplyConfiguration{{
+					Name: ptr.To("redpanda"),
+					Env: []applycorev1.EnvVarApplyConfiguration{
+						*applycorev1.EnvVar().WithName("__REDPANDA_DISABLE_BUILTIN_TRIAL_LICENSE").WithValue("true"),
+					},
+				}},
+			},
+		},
+	}
+	rp.Spec.ClusterSpec.Enterprise = &redpandav1alpha2.Enterprise{
+		License: &license,
+	}
+
+	var licenseStatus *redpandav1alpha2.RedpandaLicenseStatus
+	s.applyAndWaitFor(func(o client.Object, err error) (bool, error) {
+		if err != nil {
+			return false, err
+		}
+		rp := o.(*redpandav1alpha2.Redpanda)
+
+		for _, cond := range rp.Status.Conditions {
+			if cond.Type == redpandav1alpha2.ClusterLicenseValid {
+				// grab the first non-unknown status
+				if cond.Status != metav1.ConditionUnknown {
+					licenseStatus = rp.Status.LicenseStatus
+					return true, nil
+				}
+				return false, nil
+			}
+		}
+		return false, nil
+	}, rp)
+
+	s.Require().NotNil(licenseStatus.Expired)
+	s.Require().False(*licenseStatus.Expired)
+	s.Require().False(licenseStatus.Violation)
+	s.Require().Equal("redpanda-testing", ptr.Deref(licenseStatus.Organization, ""))
+	s.Require().Equal("enterprise", ptr.Deref(licenseStatus.Type, ""))
```

Comment:
This would be more clear if you just compared the struct (Assuming we can set all the values):
```suggestion
	s.Require().Equal(licenseStatus, redpandav1aplha2.LicenseStatus{
	    Expired: ptr.To(false),
	    Violation: false,
	    /* etc ... */
	    /* If there's fields that we can't assert with statically, just include them:*/
	    Expiry: licenseStatus.Expiry,
	})
```